miden-crypto/src/dsa/rpo_falcon512/keys/secret_key.rs

```diff
@@ -225,6 +228,30 @@ impl SecretKey {
             }
         }
     }
+
+    /// Deterministically generates a seed for seeding the PRNG used in the trapdoor sampling
+    /// algorithm used during signature generation.
+    ///
+    /// This uses the argument described in [RFC 6979](https://datatracker.ietf.org/doc/html/rfc6979#section-3.5)
+    /// § 3.5 where the concatenation of the private key and the hashed message, i.e., sk || H(m),
+    /// is used in order to construct the initial seed of a PRNG. See also [1].
+    ///
+    ///
+    /// Note that we hash in also a `log_2(N)` where `N = 512` in order to domain separate between
+    /// different versions of the Falcon DSA, see [1] Section 3.4.1.
+    ///
+    /// [1]: https://github.com/algorand/falcon/blob/main/falcon-det.pdf
+    #[cfg(feature = "std")]
+    fn generate_seed(&self, message: &Word) -> [u8; 32] {
```

Comment:
Similar to the above comment, does this need to be behind `#[cfg(feature = "std")]` flag?